hibernate-core/src/main/java/org/hibernate/bytecode/internal/bytebuddy/BytecodeProviderImpl.java

```diff
@@ -38,8 +38,8 @@
 public class BytecodeProviderImpl implements BytecodeProvider {
 
 	private static final ByteBuddyState bytebuddy = new ByteBuddyState();
-	private static final NamingStrategy.SuffixingRandom instantiatorName = new NamingStrategy.SuffixingRandom( "HibernateInstantiator" );
-	private static final NamingStrategy.SuffixingRandom optimizerName = new NamingStrategy.SuffixingRandom( "HibernateAccessOptimizer" );
+	private static final String INSTANTIATOR_PROXY_NAMING_SUFFIX = "HibernateInstantiator";
+	private static final String OPTIMIZER_PROXY_NAMING_SUFFIX = "HibernateAccessOptimizer";
```

Comment:
Just wondering if this was necessary, we allocate millions of these NamingStrategy instances so you're undoing some previous fixes.

Reply:
See my answer above.

I'm a bit surprised by the numbers though? Millions, really?

At the moment, the classes were created as `ReflectionOptimizer.InstantiationOptimizer` which is a bit weird if you ask me. And with the lookup method, we need the class to end up in the original class package.

Reply:
Because of a weird interaction of testing helpers and interceptors defined on the Suite, each test will trigger an enhancement attempt of many classes on the classpath - even things which have already been enhanced in a previous test run. So you have test X classes = millions.

Luis is looking at that but it's complex. For now, I just tried to make the enhancement as cheap as possible :)

Reply:
Yeah, I'm all to reduce allocations and object creations, we fought them hard in HV. Unfortunately, for now, I don't see another option.